cmd/node_agent/na/nodeagent.go

```diff
@@ -163,12 +165,20 @@ func (na *nodeAgentInternal) createRequest() ([]byte, *pb.Request, error) {
 		Org:        na.config.ServiceIdentityOrg,
 		RSAKeySize: na.config.RSAKeySize,
 	})
-
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to generate CSR: %v", err)
 	}
 
-	return privKey, &pb.Request{CsrPem: csr}, nil
+	var cred []byte
+	cred, err = na.pr.GetAgentCredential()
```

Comment:
why not just `cred, err :=`?

Reply:
will change it. My editor complained about this at first so I wrote it this way..